src/main_thread/decrypt/find_key_system.ts

```diff
@@ -88,52 +88,71 @@ interface IKeySystemType {
 }
 
 /**
- * @param {Object} keySystem
- * @param {Object} askedConfiguration
+ * @param {Object} newConfiguration
+ * @param {Object} prevConfiguration
  * @param {MediaKeySystemAccess} currentKeySystemAccess
- * @param {Object} currentKeySystemOptions
  * @returns {null|Object}
  */
 function checkCachedMediaKeySystemAccess(
-  keySystem: IKeySystemOption,
-  askedConfiguration: MediaKeySystemConfiguration,
+  newConfiguration: MediaKeySystemConfiguration,
+  prevConfiguration: MediaKeySystemConfiguration,
   currentKeySystemAccess: MediaKeySystemAccess | ICustomMediaKeySystemAccess,
-  currentKeySystemOptions: IKeySystemOption,
-): null | {
-  keySystemOptions: IKeySystemOption;
-  askedConfiguration: MediaKeySystemConfiguration;
-  keySystemAccess: MediaKeySystemAccess | ICustomMediaKeySystemAccess;
-} {
+): null | MediaKeySystemAccess | ICustomMediaKeySystemAccess {
   const mksConfiguration = currentKeySystemAccess.getConfiguration();
   if (shouldRenewMediaKeySystemAccess() || isNullOrUndefined(mksConfiguration)) {
     return null;
   }
 
-  // TODO Do it with MediaKeySystemAccess.prototype.keySystem instead
-  if (keySystem.type !== currentKeySystemOptions.type) {
+  if (newConfiguration.label !== prevConfiguration.label) {
     return null;
   }
 
-  if (
-    (!isNullOrUndefined(keySystem.persistentLicenseConfig) ||
-      keySystem.persistentState === "required") &&
-    mksConfiguration.persistentState !== "required"
-  ) {
+  const prevDistinctiveIdentifier = prevConfiguration.distinctiveIdentifier ?? "optional";
+  const newDistinctiveIdentifier = newConfiguration.distinctiveIdentifier ?? "optional";
+  if (prevDistinctiveIdentifier !== newDistinctiveIdentifier) {
     return null;
   }
 
-  if (
-    keySystem.distinctiveIdentifier === "required" &&
-    mksConfiguration.distinctiveIdentifier !== "required"
-  ) {
+  const prevPersistentState = prevConfiguration.persistentState ?? "optional";
+  const newPersistentState = newConfiguration.persistentState ?? "optional";
+  if (prevPersistentState !== newPersistentState) {
     return null;
   }
 
-  return {
-    keySystemOptions: keySystem,
-    keySystemAccess: currentKeySystemAccess,
-    askedConfiguration,
-  };
+  const prevInitDataTypes = prevConfiguration.initDataTypes ?? [];
+  const newInitDataTypes = newConfiguration.initDataTypes ?? [];
+  if (!isArraySubsetOf(newInitDataTypes, prevInitDataTypes)) {
+    return null;
+  }
+
+  const prevSessionTypes = prevConfiguration.sessionTypes ?? [];
+  const newSessionTypes = newConfiguration.sessionTypes ?? [];
+  if (!isArraySubsetOf(newSessionTypes, prevSessionTypes)) {
+    return null;
+  }
```

Comment:
I think `prevConfiguration` will still be the old `["temporary", "persistant"]` one in this case

Reply:
Maybe we can add a unit test for `getMediaKeySystemAccess` and check that it returns correctly an event of type `IReuseMediaKeySystemAccessEvent` in this case?

Reply:
I think it may be a biggest bang for the buck if I check for `MediaKeySytemAccess` reusage in various `keySystems` configuration in an integration test as allowed here #1478

I mean we could do both kinds of tests and it would be theoretically better, but `decrypt/` code changes often and I don't like adding supplementary friction here.

We could also just test the "should we reuse cache" function, which would not need any mock and thus not have the issues of testing `getMediaKeySystemAccess` (which requires mocking `requestMediaKeySystemAccess` and its return value, `MediaKeySystemAttacher`, and maybe some other utils), but it seems simple enough to understand the various cases that here I'm unsure of the benefits.

Reply:
Here's an example of how those integration tests could be implemented:
https://github.com/canalplus/rx-player/pull/1478/commits/c2a0d34f17b680e14119ff08eeef77c0528f3873

I did not add tests for everything yet, it's just to show you what I mean here.

Reply:
I tried on my own to make a test, it needs to mock the `requestMediaKeySystemAccess` API,  I think this is reasonably simple:
https://github.com/canalplus/rx-player/pull/1603